### PR TITLE
Windows warnings

### PR DESCRIPTION
**Pull request by raver119**
***WIP; DO NOT MERGE;***

This PR removes lots of build warnings on Windows + MSVC.
On top of that, this PR changes Windows build system to Ninja, to support parallel compilation on Windows.

### CODE REVIEW

File: libnd4j/buildnativeoperations.sh [hunk 1]
```diff
@@ -265,12 +265,18 @@ case "$OS" in
...
+        export CMAKE_COMMAND="cmake -G \"Ninja\""
+        export MAKE_COMMAND="ninja"
+        export CC="cl.exe"
+        export CXX="cl.exe"
+        PARALLEL="true"
     else
         export CMAKE_COMMAND="cmake -G \"MSYS Makefiles\""
         export MAKE_COMMAND="make"
+
+        # Sam, do we really need this?
```
Comment (saudet) [inline libnd4j/buildnativeoperations.sh]: Probably not... BTW. why not use ninja for everything, not just Windows CUDA?

Comment (raver119) [reply libnd4j/buildnativeoperations.sh]: I was solving 1 particular problem: CUDA compilation time on Windows.
For other platforms we have regular make, which works just fine afaik. Only Windows+CUDA had sequential builds.

Comment (raver119) [reply libnd4j/buildnativeoperations.sh]: p.s. Ninja is available as package for MinGW.

Comment (saudet) [reply libnd4j/buildnativeoperations.sh]: Meh, I guess we can switch all to ninja gradually :) @sshepel ?

Comment (raver119) [reply libnd4j/buildnativeoperations.sh]: Well, if build script code will be the same for all platforms - that would be good definitely. We just need to make sure there's no issues with Ninja availability on other platforms: Linux, macOS.

**Comment by saudet** [comment]
Yes, pretty sure, it's in the release notes as I wrote here: https://github.com/deeplearning4j/deeplearning4j/issues/5229#issuecomment-390096228

**Pull request merged**